[x86] Synchronize fp128 and i128 ABI tests (nfc)

tgross35 opened the pull request:
The ABI tests for fp128 covers a handful of things that the i128 test currently does not, including targets other than x86-64 Linux and some simpler argument cases. Update the i128 test to be consistent with fp128. All existing test function patterns still exist, though they all wind up renamed.

Additionally correct some `call` signatures in the fp128 test (not relevant for output).

tgross35 committed:
[x86] Combine fp128 and i128 ABI tests (nfc)

The ABI tests for fp128 covers a handful of things that the i128 test
currently does not, including targets other than x86-64 Linux and some
simpler argument cases. Since these tests should test the same
function patterns, and since some of the directives can be reused,
combine `fp128-abi.ll` and `i128-abi.ll` into one test using `sed` to
set the type.

phoebewang commented:
How about replace the type in the RUN command, so that we can share the same source and probably same CHECK when you fixed it. An example in llvm/test/CodeGen/X86/vector-lrint-f16.ll

tgross35 committed:
[x86] Rename the fp128 ABI test to i128-fp128-abi (nfc)

This test will be updated to cover both fp128 and i128. Do the rename in
a separate commit to avoid confusing git's similarity index.

tgross35 commented:
Great idea, did what you suggested and it's pretty clean. A second `sed` was needed to use either `0xL0` or `0` for a literal zero, not sure if there is something better to do here.

Renamed the file in the first commit so git history correctly reflects that the file is modified rather than deleted+re-added. If that's important, the first commit will need to be pushed before this PR merges.

phoebewang commented on llvm/test/CodeGen/X86/i128-fp128-abi.ll:
```diff
@@ -0,0 +1,903 @@
+; CHECK-X86-LABEL: return:
+; CHECK-X86:       # %bb.0:
+; CHECK-X86-NEXT:    pushl %edi
+; CHECK-X86-NEXT:    .cfi_def_cfa_offset 8
```
Add `nounwind` to remove such directives.

tgross35 replied:
Thanks, added the `nounwind`s.

I'll need you to land this for me (note the git quirk at https://github.com/llvm/llvm-project/pull/147923#issuecomment-3059835114)

phoebewang reviewed (approved):
LGTM with one nit.

tgross35 committed:
add nounwind
